service/src/main.rs

```diff
@@ -404,22 +415,55 @@ fn start_worker<E, T, D>(
 	}
 }
 
-/// Triggers the enclave to produce a block based on a fixed time schedule
+/// Triggers the enclave to produce a block based on a fixed time schedule.
 fn start_interval_block_production<E: EnclaveBase + SideChain>(enclave_api: &E) {
 	use itp_settings::sidechain::SLOT_DURATION;
 
-	let mut interval_start = SystemTime::now();
-	loop {
-		if let Ok(elapsed) = interval_start.elapsed() {
-			if elapsed >= SLOT_DURATION {
-				// update interval time
-				interval_start = SystemTime::now();
-				execute_trusted_operations(enclave_api);
-			} else {
-				// sleep for the rest of the interval
-				let sleep_time = SLOT_DURATION - elapsed;
-				thread::sleep(sleep_time);
+	schedule_on_repeating_intervals(
+		|| {
+			execute_trusted_calls(enclave_api);
+		},
+		SLOT_DURATION,
+	);
+}
+
+/// Starts the execution of trusted getters in repeating intervals.
+///
+/// The getters are executed in a pre-defined slot duration.
+fn start_interval_trusted_getter_execution<E: SideChain>(enclave_api: &E) {
+	use itp_settings::enclave::TRUSTED_GETTERS_SLOT_DURATION;
+
+	schedule_on_repeating_intervals(
+		|| {
+			if let Err(e) = enclave_api.execute_trusted_getters() {
+				error!("Execution of trusted getters failed: {:?}", e);
 			}
+		},
+		TRUSTED_GETTERS_SLOT_DURATION,
+	);
+}
+
+/// Schedules a task on perpetually looping intervals.
+///
+/// In case the task takes longer than is scheduled by the interval duration,
+/// the interval timing will drift. The task is responsible for
+/// ensuring it does not use up more time than is scheduled.
+fn schedule_on_repeating_intervals<T>(task: T, interval_duration: Duration)
+where
+	T: Fn(),
+{
+	let mut interval_start = Instant::now();
+	loop {
+		let elapsed = interval_start.elapsed();
+
+		if elapsed >= interval_duration {
+			// update interval time
+			interval_start = Instant::now();
+			task();
+		} else {
+			// sleep for the rest of the interval
+			let sleep_time = interval_duration - elapsed;
+			thread::sleep(sleep_time);
 		}
 	}
```

Comment:
Extracted a function to run a task on a scheduled interval, repeating perpetually